cleanlab_studio/internal/util.py

```diff
@@ -13,16 +12,18 @@
 
 
 DatasetSourceType = TypeVar(
-    "DatasetSourceType", bound=Union[str, os.PathLike, pd.DataFrame, pyspark.sql.DataFrame]
+    "DatasetSourceType", bound=Union[str, pathlib.Path, pd.DataFrame, pyspark.sql.DataFrame]
 )
 
 
 def init_dataset_source(
     dataset_source: DatasetSourceType, dataset_name: Optional[str] = None
 ) -> DatasetSource:
     if isinstance(dataset_source, pd.DataFrame):
+        assert dataset_name is not None
         return PandasDatasetSource(df=dataset_source, dataset_name=dataset_name)
     elif isinstance(dataset_source, pyspark.sql.DataFrame):
+        assert dataset_name is not None
```

Comment:
this is a user-facing error, so let's give a good error message. smthn like:

`dataset_name` must be provided for `dataset_source` of type `(pd|pyspark.sql).DataFrame`